packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.views.text.frescosupport
+
+import android.content.res.Resources
+import android.graphics.BlendMode
+import android.graphics.BlendModeColorFilter
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.PorterDuff
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.os.Build
+import android.widget.TextView
+import com.facebook.drawee.controller.AbstractDraweeControllerBuilder
+import com.facebook.drawee.generic.GenericDraweeHierarchy
+import com.facebook.drawee.generic.GenericDraweeHierarchyBuilder
+import com.facebook.drawee.view.DraweeHolder
+import com.facebook.imagepipeline.request.ImageRequest
+import com.facebook.imagepipeline.request.ImageRequestBuilder
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.modules.fresco.ReactNetworkImageRequest
+import com.facebook.react.uimanager.PixelUtil
+import com.facebook.react.views.image.ImageResizeMode
+import com.facebook.react.views.text.internal.span.TextInlineImageSpan
+
+/**
+ * FrescoBasedTextInlineImageSpan is a span for Images that are inside <Text/>. It computes its size
+ * based on the input size. When it is time to draw, it will use the Fresco framework to get the
+ * right Drawable and let that draw.
+ *
+ * Since Fresco needs to callback to the TextView that contains this, in the ViewManager, you must
+ * tell the Span about the TextView
+ *
+ * Note: It borrows code from DynamicDrawableSpan and if that code updates how it computes size or
+ * draws, we need to update this as well.
+ */
+public class FrescoBasedReactTextInlineImageSpan(
```

Comment:
@mateoguzmana The CI build failed. Maybe the class should've been private?

```
> Task :packages:react-native:ReactAndroid:configureCMakeRelWithDebInfo[arm64-v8a]
e: file:///__w/react-native/react-native/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/frescosupport/FrescoBasedReactTextInlineImageSpan.kt:53:5 Subclass 'public' exposes its 'internal' supertype 'TextInlineImageSpan'.
```

Reply:
Was just checking this, the direct conversion from Java would be to make this class internal:

```suggestion
internal class FrescoBasedReactTextInlineImageSpan(
```

Reply:
Alright, thanks!